pkg/ring/replication_strategy.go

```diff
@@ -2,51 +2,51 @@ package ring
 
 import (
 	"fmt"
+	"time"
 )
 
-// replicationStrategy decides, given the set of ingesters eligible for a key,
-// which ingesters you will try and write to and how many failures you will
-// tolerate.
-// - Filters out dead ingesters so the one doesn't even try to write to them.
-// - Checks there is enough ingesters for an operation to succeed.
-// The ingesters argument may be overwritten.
-func (r *Ring) replicationStrategy(ingesters []IngesterDesc, op Operation) ([]IngesterDesc, int, error) {
+// replicationStrategy decides, given the set of tokens eligible for a key,
+// which tokens will be used for an operation and how many failures will
+// be tolerated. Tokens is expected to be a list where each token belongs
+// to a unique ingester.
+//
+// replicationStrategy will check to see if there's enough tokens for
+// an operation to succeed; unhealthy tokens are filtered out using
+// IsHealthyState.
+func (r *Ring) replicationStrategy(tokens []TokenDesc, op Operation) ([]TokenDesc, int, error) {
```

Comment:
I would love to remove this from `*Ring`, either to be standalone function, or move to `*Desc` instead. It only uses `r.cfg.ReplicationFactor` from `r`. [Optional]